studybuddies/src/app/components/signin/signin.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component, OnInit } from '@angular/core';
+import { AuthService } from '../../services/auth.service'
+import { Router } from '@angular/router'
+
+@Component({
+  selector: 'app-signin',
+  templateUrl: './signin.component.html',
+  styleUrls: ['./signin.component.css']
+})
+export class SigninComponent implements OnInit {
+
+  user = {
+    email: '',
+    password: ''
+  }
+
+  constructor(
+    private authService: AuthService,
+    private router: Router
+    ) {}
+
+  ngOnInit(): void {
+  }
+
+  signIn() {
+    this.authService.signIn(this.user)
+      .subscribe(
+        res => {
+          console.log(res);
+          localStorage.setItem('token', res.token);
+          this.router.navigate(['/posts']);
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Frontend/pullRequest?prid=7163724)
```suggestion
          this.router.navigate(["/posts"]);
```